sdk/keyvault/keyvault-keys/test/internal/apiVersionParameter.spec.ts

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import * as assert from "assert";
+import { env, Recorder } from "@azure/test-utils-recorder";
+import { createSandbox } from "sinon";
+
+import { KeyClient } from "../../src";
+import { authenticate } from "../utils/testAuthentication";
+import TestClient from "../utils/testClient";
+import { URLBuilder } from "@azure/core-http";
+import { ClientSecretCredential } from "@azure/identity";
+
+describe("The Keys client should set the apiVersion", () => {
+  const secretPrefix = `apiVersion${env.KEY_NAME || "KeyName"}`;
+  let secretSuffix: string;
+  let recorder: Recorder;
+  let testClient: TestClient;
+  let keyVaultUrl: string;
+  let credential: ClientSecretCredential;
+
+  beforeEach(async function() {
+    const authentication = await authenticate(this);
+    secretSuffix = authentication.secretSuffix;
+    recorder = authentication.recorder;
+    testClient = authentication.testClient;
+    keyVaultUrl = authentication.keyVaultUrl;
+    credential = authentication.credential;
+  });
+
+  afterEach(async function() {
+    await recorder.stop();
+  });
+
+  // The tests follow
+
+  it("it should default to the latest API version", async function() {
+    const sandbox = createSandbox();
+    const spy = sandbox.spy(URLBuilder.prototype, "toString");
```

Comment:
feels weird to spy on URLBuilder in this way- also `URLBuilder` is getting removed in v2

Reply:
If all you care is that we can set the api-version, maybe just put a custom mock httpclient interface that looks at what the query param value is?

Reply:
I'll take a different approach then!